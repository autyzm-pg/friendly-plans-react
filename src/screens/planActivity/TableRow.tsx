import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

import { CheckboxInput, Icon, IconButton } from 'components';
import { PlanItem } from 'models';
import { NavigationService } from 'services';
import { palette, typography } from 'styles';
import { CELL_DIMENSIONS } from './TaskTable';

interface Props {
  rowNumber: number;
  border?: boolean;
  planItem: PlanItem;
}

export const TableRow: React.FunctionComponent<Props> = ({ rowNumber, planItem, border }) => {
  const navigateToPlanItemUpdate = () => {
    NavigationService.navigate('PlanItemTask', {
      planItem,
    });
  };

  const onDelete = () => {
    planItem.delete();
  };

  const handleCheckboxChange = () => {
    planItem.setComplete(!planItem.completed);
  };

  return (
    <View style={[styles.row, border && styles.rowBorder]}>
      <View style={[styles.cell, { flex: CELL_DIMENSIONS.NUMBER, alignItems: 'center' }]}>
        <Text style={styles.text}>{rowNumber}</Text>
      </View>
      <View style={[styles.cell, { flex: CELL_DIMENSIONS.CHECKBOX, alignItems: 'center' }]}>
        <CheckboxInput checked={planItem.completed} onPress={handleCheckboxChange} />
      </View>
      <View style={[styles.cell, { flex: CELL_DIMENSIONS.NAME }]}>
        <Text style={styles.text}>{planItem.name}</Text>
      </View>
      <View style={[styles.cell, { flex: CELL_DIMENSIONS.TYPE, alignItems: 'flex-start' }]}>
        <Icon name={planItem.getIconName()} type="material" />
      </View>
      <View style={[styles.cell, { flex: CELL_DIMENSIONS.SECTION }]}>
        <Text style={styles.text}>5</Text>
      </View>
      <View style={[styles.cell, { flex: CELL_DIMENSIONS.TIME }]}>
        <Text style={styles.text}>{planItem.time}</Text>
      </View>
      <View style={[styles.cell, { flex: CELL_DIMENSIONS.DELETE, alignItems: 'flex-start' }]}>
        <IconButton name="delete" size={24} color={palette.textInputPlaceholder} onPress={onDelete} />
      </View>
      <View style={[styles.cell, { flex: CELL_DIMENSIONS.EDIT, alignItems: 'center' }]}>
        <IconButton name="pencil" size={24} onPress={navigateToPlanItemUpdate} />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  row: {
    height: 56,
    alignSelf: 'stretch',
    flexDirection: 'row',
  },
  rowBorder: {
    borderBottomColor: palette.backgroundAdditional,
    borderBottomWidth: 1,
  },
  text: {
    ...typography.body,
    color: palette.textBody,
  },
  cell: {
    alignSelf: 'stretch',
    justifyContent: 'center',
  },
});
