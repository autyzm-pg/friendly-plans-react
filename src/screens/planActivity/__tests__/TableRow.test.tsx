import { shallow } from 'enzyme';
import React from 'react';

import { TableRow } from '../TableRow';

describe('TableRow', () => {
  it('renders correctly', () => {
    const tableRow = shallow(<TableRow rowNumber={1} />);

    expect(tableRow).toMatchSnapshot();
  });
});
