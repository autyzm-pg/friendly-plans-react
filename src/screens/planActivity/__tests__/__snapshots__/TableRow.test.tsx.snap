// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "flexDirection": "row",
        "height": 56,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "alignItems": "center",
          "flex": 0.3,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#3C35A8",
          "fontFamily": "Rubik-Regular",
          "fontSize": 14,
          "letterSpacing": 0,
        }
      }
    >
      1
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "flex": 0.3,
        },
      ]
    }
  >
    <CheckboxInput
      checked={true}
      onPress={[Function]}
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "flex": 1.7,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#3C35A8",
          "fontFamily": "Rubik-Regular",
          "fontSize": 14,
          "letterSpacing": 0,
        }
      }
    >
      Bębenki afrykańskie
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "alignItems": "flex-start",
          "flex": 0.5,
        },
      ]
    }
  >
    <Icon
      name="layers"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "flex": 0.5,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#3C35A8",
          "fontFamily": "Rubik-Regular",
          "fontSize": 14,
          "letterSpacing": 0,
        }
      }
    >
      5
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "flex": 1.9,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#3C35A8",
          "fontFamily": "Rubik-Regular",
          "fontSize": 14,
          "letterSpacing": 0,
        }
      }
    >
      15'
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "alignItems": "flex-start",
          "flex": 0.4,
        },
      ]
    }
  >
    <IconButton
      color="#C8CBFA"
      name="delete"
      size={24}
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "alignItems": "center",
          "flex": 0.6,
        },
      ]
    }
  >
    <IconButton
      name="pencil"
      size={24}
    />
  </View>
</View>
`;
